libraries/botbuilder/src/skills/skillHandler.ts

```diff
@@ -31,6 +31,14 @@ import {
 import { ChannelServiceHandler } from '../channelServiceHandler';
 import { BotFrameworkAdapter } from '../botFrameworkAdapter';
 
+/**
+ * Casts adapter to BotFrameworkAdapter only if necessary
+ * @param adapter adapter to maybe cast as BotFrameworkAdapter
+ */
+function maybeCastAdapter(adapter: BotAdapter): BotFrameworkAdapter {
```

Comment:
@johnataylor created https://github.com/microsoft/botbuilder-js/issues/2935 to track fixing the `BotAdapter` class.

Reply:
Sounds reasonable, although I hate to drag the past forward with us in light of the CloudAdapter work underway.